ESP8266/ESP8266.cpp

```diff
@@ -181,7 +182,13 @@ int8_t ESP8266::getRSSI()
 
 bool ESP8266::isConnected(void)
 {
-    return getIPAddress() != 0;
+    const char *ip_buff = getIPAddress();
+
+    if(!ip_buff || std::strcmp(ip_buff, "0.0.0.0") == 0) {
+        return 0;
+    }
+
+    return ip_buff;
```

Comment:
Returning const char * in a function that should return bool?

Reply:
Heh. That does actually work - pointers quietly convert to "true" if non-NULL, "false" if NULL, in both C++ and C99. But probably not what we want to do here.

Reply:
Agree